web/modules/custom/product_uploader/product_uploader.module

```diff
@@ -0,0 +1,27 @@
+<?php
+
+/**
+ * @file
+ * Contains hook for Product Uploader module.
+ */
+
+declare(strict_types=1);
+
+use Drupal\Core\Entity\EntityInterface;
+
+/**
+ * Implements hook_entity_presave().
+ */
+function product_uploader_entity_presave(EntityInterface $entity) {
```

Comment:
As this hook is called when any kind of entity is created or update, you should rather use

https://api.drupal.org/api/drupal/core%21lib%21Drupal%21Core%21Entity%21entity.api.php/function/hook_ENTITY_TYPE_presave/8.8.x

This only gets called when the entity of an entity type is created or updated. This way you can also remove the outermost condition from the hook.

Reply:
I'm going to need some help with this.